### PR TITLE
Add X-Cache header for http queries

### PR DESCRIPTION
**Pull request by sgrebnov**
Part of https://github.com/spiceai/spiceai/issues/1358

Add `X-Cache` header to indicate whether query result is from results cache (`hit`) or not(`miss`). Applied only when results cache is enabled.

### CODE REVIEW

**Review by digadeesh — DISMISSED** [review]

**Review by digadeesh — DISMISSED** [review]

File: crates/cache/src/lib.rs [hunk 1]
```diff
@@ -44,6 +45,24 @@ pub enum Error {
 
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
+pub enum ResultCacheStatus {
+    Cached,
```
Comment (lukekim) [inline crates/cache/src/lib.rs:49]: Let's call this Hit/Miss

File: crates/cache/src/lib.rs [hunk 2]
```diff
@@ -44,6 +45,24 @@ pub enum Error {
...
+    NotCached,
+    CacheDisabled,
+}
```
Comment (lukekim) [inline crates/cache/src/lib.rs]:
```suggestion
pub enum RetrievalResult {
    Hit,
    Miss
}
```

Don't put disabled in here.

File: crates/runtime/src/datafusion.rs [hunk 3]
```diff
@@ -267,7 +269,7 @@ impl DataFusion {
         &self,
         sql: &str,
         restricted_sql_options: Option<SQLOptions>,
-    ) -> Result<SendableRecordBatchStream> {
+    ) -> Result<QueryResult> {
```
Comment (lukekim) [inline crates/runtime/src/datafusion.rs:270]: Make this Optional. If it is missing, cache is disabled.

Comment (sgrebnov) [reply crates/runtime/src/datafusion.rs:270]: Like the proposed approach with Option< RetrievalResult>. Updated

File: crates/cache/src/lib.rs (hunk 2 above)
Comment (sgrebnov) [reply crates/cache/src/lib.rs]: Like the proposed approach with Option< RetrievalResult>. Updated

**Review by lukekim — APPROVED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**